File: packages/superwall/android/src/main/java/io/capawesome/capacitorjs/plugins/superwall/classes/results/GetUserIdResult.java
package io.capawesome.capacitorjs.plugins.superwall.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.superwall.interfaces.Result;

public class GetUserIdResult implements Result {

    @NonNull
    private final String userId;

    public GetUserIdResult(@NonNull String userId) {
        this.userId = userId;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject jsObject = new JSObject();
        jsObject.put("userId", userId);
        return jsObject;
    }
}

File: packages/superwall/ios/Plugin/Classes/Results/GetUserIdResult.swift
import Foundation
import Capacitor

@objc public class GetUserIdResult: NSObject, Result {
    let userId: String

    init(userId: String) {
        self.userId = userId
    }

    @objc public func toJSObject() -> AnyObject {
        var jsResult = JSObject()
        jsResult["userId"] = userId
        return jsResult as AnyObject
    }
}
